### PR TITLE
Fix for python 3.6.0, 3.6.1

### PR DESCRIPTION
**Pull request by farizrahman4u**
image

### CODE REVIEW

File: wandb/sdk/wandb_run.py [hunk 1]
```diff
@@ -62,10 +62,15 @@
...
         Union,
-        NoReturn,
         Type,
         Callable,
     )
+
+    try:
+        from typing import NoReturn
+    except ImportError:
+        NoReturn = None  # type: ignore
```
Comment (raubitsj) [inline wandb/sdk/wandb_run.py:72]: Just change the NoReturn later on to "NoReturn"

**Review by raubitsj — CHANGES_REQUESTED** [review]
Change NoReturn to "NoReturn" is a cleaner solution for now.

File: wandb/sdk/wandb_run.py (hunk 1 above)
Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:72]: Do you mean like:
```python
def exit(self, code: object = 0) -> "NoReturn":
     #####
```

That fails with `error: Name 'NoReturn' is not defined`

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: That is correct,  with python3.0/3.1 that code should never be executed as it is converted to:
wandb/sdk_py27/wandb_run.py
which strips the annotation.

Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:72]: Yes, but that still cause flake8 and mypy checks to fail with `error: Name 'NoReturn' is not defined`.

Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:72]: Nvm, I'll try ignoring those

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: We only should be running flake8/mypy on python3.6+
how are you getting errors?

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: Maybe we are missing a specifier in tox.ini saying that py3.6 is the requirement, i think it just says python3 and it normally picks up py3.6+
noone should be using python 3.6.0/1. it is often restricted from setup.py.

Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:72]: image

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: heh, thanks, i guess there are 3.6.1 users.. they arent using modern pytorch or tensorflow as that support was dropped year+ ago?

ok, lets keep 3.6.1 support, but the string annotation should just work.  our library runtime will work fine on 3.6.1 but our flake8 and mypy checking should only be run for 3.6.2+
See: https://pypi.org/project/torch/

Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:72]: The string annotation will only work if there is actually a variable with the same name, else flake8 and mypy complains (`tox -e flake8`, `tox -e mypy`). I'm not running the tests on 3.6.1.

Checkout this diff:
https://github.com/wandb/client/commit/dcbeda7267fa5ec40132d173a152538e37b82ace

And the CI output:
https://app.circleci.com/pipelines/github/wandb/client/3256/workflows/91f0385d-513c-45bf-8d24-aaaee0c4fd79/jobs/37119

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: Here is the PR I was thinking would be better:
https://github.com/wandb/client/pull/1848

Comment (raubitsj) [reply wandb/sdk/wandb_run.py:72]: This makes NoReturn only used during typechecking (mypy).  runtime 3.6.0, 3.6.1 should be fine.

**Comment by raubitsj** [comment]
https://github.com/wandb/client/pull/1848 merged which solves the same issue.

**Pull request closed without merging**